[68459] Add a migration to ensure anonymous user has persisted preferences and theme set to sync_with_os

bsatarnejad opened the pull request:
# Ticket
https://community.openproject.org/wp/68459

bsatarnejad committed:
ensure anonymous user has persisted preferences and theme set to sync_with_os

oliverguenther committed:
Rewrite migration using UPSERT, and ensure uniqueness constraint on user_id

myabc commented on db/migrate/20251105082430_ensure_anonymous_user_preference_exists_and_set_theme.rb:
```diff
@@ -0,0 +1,29 @@
+
+    execute <<~SQL.squish
+      UPDATE user_preferences
+      SET settings = settings || '{"theme": "light"}'::jsonb,
```
Nitpick: This value was already `sync_with_os` in my local setup prior to running this migration. So `./bin/rake db:rollback STEP=2` isn't strictly rolling back. However, since we're not journaling `UserPreference`, I guess there's a clean solution to this.

myabc reviewed (changes requested):
Tested locally - works for me! Also ran `bundle exec rake active_record_doctor` in case i missed anything.

There are two spec failures. However, once those are fixed, should be good to merge.

```
rspec ./spec/services/user_preferences/update_service_integration_spec.rb:77
```

cbliard committed:
Fix failing spec

There is a unique constraint on `user_preference.user_id`, so it raised
an error when trying to create a `UserPreference` for `current_user` as
it already has one.

cbliard commented:
The other failing feature test `./modules/meeting/spec/features/structured_meetings/work_package_meetings_tab_spec.rb:361` is passing locally.
Merging.
